graph/number_of_islands_dfs.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file
+ *
+ * @brief Algorithm to count the number of islands
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂